codec-memcache/src/main/java/io/netty/handler/codec/memcache/binary/DefaultFullBinaryMemcacheRequest.java

```diff
@@ -113,6 +113,36 @@ public FullBinaryMemcacheRequest duplicate() {
         if (extras != null) {
             extras = extras.duplicate();
         }
-        return new DefaultFullBinaryMemcacheRequest(key, extras, content().duplicate());
+        return copy(key, extras, content().duplicate());
+    }
+
+    @Override
+    public FullBinaryMemcacheRequest retainedDuplicate() {
+        ByteBuf key = key();
```

Comment:
`return replace(key, extras, content().retainedDuplicate());` ... assuming `replace` should also be calling `retainedDuplicate()` on `key` and `extras`.